### PR TITLE
Reorganize the files to easily generate the plugin with ruby gems

### PR DESCRIPTION
**Pull request by jonathanlbt1**
# Pull Request

## Description

In this pull request (PR), I reorganized the files to create the “fluent-plugin-openlineage” plugin, designed to work alongside Fluentd. Specifically:

### Generate the Plugin

In order to document how to use this plugin, I made the necessary changes below:

 - adjusted the README file.
 - added previous relevant information to a separate file called PROJECT_NOTES.md.

I recommend that someone from the OpenLineage team generate the plugin and upload it to rubygems.org, keeping it within the OpenLineage ecosystem. For instructions on how to create a gem, refer to this link: [Make Your Own Gem](https://guides.rubygems.org/make-your-own-gem/).

To facilitate easy installation of this plugin with Fluentd, follow the standard naming convention: prefix the plugin name with fluent-plugin-.... By doing so, you can install it using the command `fluent-gem install fluent-plugin-openlineage`, allowing Fluentd to promptly integrate the plugin into its core. Additional details can be found in the updated README file.

Lastly, I renamed the folder from fluentd to fluent-plugin-openlineage. This consistency will help others quickly identify the project’s purpose, as other plugins adhere to this naming convention.

## How This Has Been Tested

For the past three weeks, these changes have been relevant to make it possible to test this plugin on the following scenarios:

  - [x] Test A: These changes have been tested locally, with Postman performance tests using production-like payloads.
  - [x] Test B: Plugin has been deployed to kubernetes on AWS and has been exposed to a staging environment, and after a sereies of tests, is now successfully running in production environment. 

## PR's Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes

If you have further questions or need additional assistance, feel free to reach out to me! 😊🚀

### CODE REVIEW

**Review by mobuchowski — CHANGES_REQUESTED** [review]
Hey @jonathanlbt1 - thanks for working on this. I think once we get rid of the duplication we'd be good regarding directory structure. @pawel-big-lebowski can you check the Ruby part?

**Comment by jonathanlbt1** [comment]
Hi @mobuchowski ! Sorry for the delay, I have been very busy these days. I've made your requested changes and also included some other config information to increase fluentd's overall performance I recently learned.

File: proxy/fluentd/docker/conf/fluent.conf [hunk 1]
```diff
@@ -18,30 +25,25 @@
     buffered true
     serializer json
     retryable_response_codes 408, 429, 500, 502, 503
-
+    # to improve performance, you can use the following commented options:
```
Comment (pawel-big-lebowski) [inline proxy/fluentd/docker/conf/fluent.conf:28]: that's a nice hint.

File: proxy/fluentd/test/plugin/test_parser_openlineage.rb [hunk 2]
```diff
@@ -1,141 +0,0 @@
-require "helper"
-require "fluent/plugin/parser_openlineage.rb"
-
-class OpenlineageParserTest < Test::Unit::TestCase
```
Comment (pawel-big-lebowski) [inline proxy/fluentd/test/plugin/test_parser_openlineage.rb:4]: What's happening with those tests? They're important.

Comment (jonathanlbt1) [reply proxy/fluentd/test/plugin/test_parser_openlineage.rb:4]: Sorry, I had mistakenly removed them when I was re-organizing the folders to generate the plugin. I've included them back in.

**Review by pawel-big-lebowski — APPROVED** [review]
Great work. 

Feel free to rename directory into `fluent-plugin-openlineage`

**Comment by pawel-big-lebowski** [comment]
Would it make sense to get rid of `fluend` directory in favour of `openlineage-fluentd-proxy`? 
This would also require changing CI -> https://github.com/OpenLineage/OpenLineage/blob/main/.circleci/continue_config.yml#L1013

File: proxy/fluentd/spec/Naming.md [hunk 3]
```diff
@@ -0,0 +1,500 @@
+# Naming
```
[the remark this answers is not in the document]
Comment (jonathanlbt1) [reply proxy/fluentd/spec/Naming.md]: I've already fixed it. Can you please check again this and other changes you requested?

**Comment by pawel-big-lebowski** [comment]
How come the PR modifies spark integration files?

**Comment by jonathanlbt1** [comment]
> Would it make sense to get rid of `fluend` directory in favour of `openlineage-fluentd-proxy`? This would also require changing CI -> https://github.com/OpenLineage/OpenLineage/blob/main/.circleci/continue_config.yml#L1013

Yes, because basically, this project is a plugin for Fluentd, and not Fluentd itself. The project is basically the same. The only difference being the name and some other comon features used in other Fluentd plugins. I'll change the referente in the continue_config.yml file.

**Comment by jonathanlbt1** [comment]
> How come the PR modifies spark integration files?

I didn't change them myself. Probably it came from some other PR's from the remote repo. I'll merge main into my branch again and see if I can fix it.

**Comment by jonathanlbt1** [comment]
I noticed I've included a test branch into this commit so I'm closing it to upload a new one with only the changes I had in mind.

**Pull request closed without merging**